### PR TITLE
nginx: add VOD-related modules from Kaltura

### PR DESCRIPTION
**Pull request by zaninime**
###### Motivation for this change

Add a collection of useful modules for video on demand distribution via nginx. I added one commit for each module.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/nginx/modules.nix [hunk 1]
```diff
@@ -462,6 +484,17 @@ in
     inputs = [ pkgs.ffmpeg_3 ];
   };
 
+  vod = {
+    src = fetchFromGitHub {
+      name = "vod";
+      owner = "kaltura";
+      repo = "nginx-vod-module";
+      rev = "e46079f51282d5a378e6911714b5f3a533bb7700";
```
Comment (ajs124) [inline pkgs/servers/http/nginx/modules.nix:492]: They actually seem to be doing releases, why not stick to that?

**Comment by ajs124** [comment]
secure-token also has a recent release, akamai-token-validate-module however does not.

File: pkgs/servers/http/nginx/modules.nix (hunk 1 above)
Comment (zaninime) [reply pkgs/servers/http/nginx/modules.nix:492]: Unfortunately they didn't release one with recent bugfixes such as https://github.com/kaltura/nginx-vod-module/commit/a5c6825c9ae595d81c143515d3c02e2202abaebe.

**Comment by zaninime** [comment]
Unfortunately, even though there are some recent releases for two of the three modules, some bugfixes and performance improvements didn't make into a release yet. See:

- https://github.com/kaltura/nginx-secure-token-module/commit/e0f5fa410b0784d8783f8c0db74020540155cab9 (only in master)
- https://github.com/kaltura/nginx-vod-module/commit/a5c6825c9ae595d81c143515d3c02e2202abaebe (only in master)

**Comment by Izorkin** [comment]
Possibly rename to nginxModules.<name>: init at ..., like as in other PR? - https://github.com/NixOS/nixpkgs/commits/master/pkgs/servers/http/nginx/modules.nix

**Comment by zaninime** [comment]
@Izorkin done.

**Review by Izorkin — APPROVED** [review]

**Comment by Izorkin** [comment]
cc @Mic92 @bhipple

**Pull request merged**